File: ui_ux/backend/models/Offer.js
const mongoose = require("mongoose");

const Schema = mongoose.Schema;

const offerSchema = new Schema({
  studentName: {
    type: String,
    required: true,
  },
  studentId: {
    type: String,
    required: true,
  },
  teacherId: {
    type: String,
    required: true,
  },
  classStudies: {
    type: String,
    required: true,
  },
  picturePath: {
    type: String,
    default: "",
  },
  location: {
    type: String,
    required: true,
  },
  note: {
    type: String,
    required: false,
  },
});
module.exports = mongoose.model("Offer", offerSchema);

File: ui_ux/lib/models/offer.dart
class Offer {
  final String studentName;
  final String studentId;
  final String teacherId;
  final String classNumber;
  final String picturePath;
  final String location;
  final String note;

  Offer({
    required this.studentName,
    required this.studentId,
    required this.teacherId,
    required this.classNumber,
    required this.picturePath,
    required this.location,
    required this.note,
  });

  factory Offer.fromJson(Map<String, dynamic> json) {
    return Offer(
      studentName: json['studentName'],
      studentId: json['studentId'],
      teacherId: json['teacherId'],
      classNumber: json['classStudies'] ?? 0,
      picturePath: json['picturePath'],
      location: json['location'],
      note: json['note'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'studentName': studentName,
      'studentId': studentId,
      'teacherId': teacherId,
      'class': classNumber,
      'location': location,
      'note': note,
    };
  }
}
